lib/services/nativescript-project-plugins-service.ts

```diff
@@ -0,0 +1,472 @@
+			let result = this.$httpClient.httpRequest(url).wait().body;
+			if(result) {
+				let npmSearchResult = JSON.parse(result).results;
+ 				plugins =_.map(npmSearchResult, (pluginResult: any) => {
+					if(pluginResult) {
+						let pluginInfo: IPluginInfoBase = {
+							Authors: pluginResult.author,
+							Name: this.getStringFromNpmSearchResult(pluginResult, "name"),
+							Identifier: this.getStringFromNpmSearchResult(pluginResult, "name"),
+							Version: this.getStringFromNpmSearchResult(pluginResult, "version"),
+							Url: this.getStringFromNpmSearchResult(pluginResult, "homepage"),
+							Platforms: [],
+							Description: this.getStringFromNpmSearchResult(pluginResult, "description"),
+							SupportedVersion: ""
+						};
+
+						return new NativeScriptPluginData(pluginInfo, PluginType.NpmPlugin, this.$project);
+					}
+
+					return null;
+				}).filter(pl => !!pl);
+			}
+
+			return plugins;
+		}).future<IPlugin[]>()();
+	}
+
+	private getStringFromNpmSearchResult(pluginResult: any, propertyName: string): string {
+		if(pluginResult && pluginResult[propertyName] && pluginResult[propertyName].length){
+			let item = _.first(pluginResult[propertyName]);
+			if(item) {
+				return item.toString();
+			}
+		}
+
+		return "";
+	}
+
+	private getTopNativeScriptNpmPackages(count: number): IFuture<IPlugin[]> {
+		return (() => {
+			let currentPage = 0;
+			let shouldBreak = false;
+			let plugins: IPlugin[] = [];
+			do {
+				let nativescriptUrl = `${NativeScriptProjectPluginsService.NPM_SEARCH_URL}/query?fields=name,version,rating&sort=rating+desc&q=keywords:nativescript+NativeScript&start=${currentPage * count}&size=${count}`;
+				let result = this.$httpClient.httpRequest(nativescriptUrl).wait().body;
+				if(result) {
+					let npmSearchResults: any[] = JSON.parse(result).results;
+					shouldBreak = !npmSearchResults.length;
+					let pluginFutures = _.map(npmSearchResults, pluginResult => this.getDataForNpmPackage(this.getStringFromNpmSearchResult(pluginResult, "name"), this.getStringFromNpmSearchResult(pluginResult, "version")));
+					Future.wait(pluginFutures);
+					let currentPlugins = _(pluginFutures)
+									.map(fut => fut.get())
+									.filter(pl => !!pl && !!pl.data && !!pl.data.Platforms && pl.data.Platforms.length > 0)
+									.slice(0, count - plugins.length)
+									.value();
+					plugins = plugins.concat(currentPlugins);
+				} else {
+					shouldBreak = true;
+				}
+				currentPage++;
+			} while (plugins.length < count && !shouldBreak);
+
+			return plugins;
+		}).future<IPlugin[]>()();
+	}
+
+	private getDataForNpmPackage(packageName: string, version?: string): IFuture<IPlugin> {
+		return ((): IPlugin => {
+			let plugin: IPlugin;
+			try {
+				let url = `${NativeScriptProjectPluginsService.NPM_REGISTRY_URL}/${packageName}`;
+				let selectedVersion = version || "latest";
```

Comment:
You don't seem to need this variable because you only ever assign a version if the original `version` variable is present.